onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp

```diff
@@ -56,6 +56,17 @@ namespace OperatorHelper
         }
     }
 
+    template <typename T>
+    void ExpandToAxes(/*inout*/ std::vector<T>& attr, std::vector<int32_t> axes, std::vector<T> expanded)
```

Comment:
Passing by value copy allocates a copy of the `std::vector` on the stack. Let's just pass a lightweight *view* of the axes via `gsl::span`.